woocommerce/woocommerce-android
# Shipping Labels Onboarding: Pre-install screen

Part of: #6587 


### Description
This PR adds the following changes:
1. Refactoring of the installation ViewModel to allow multiple states.
2. Adds pre-install screen.
3. Handle animation between the CTA and Pre-install screen.

### Testing instructions
1. Make sure your store is in US, and uses USD as the currency, but doesn't have WCShip.
2. Create an order eligible for shipping label creation (ie: not virtual/downloadable and not eligible for IPP).
3. Open the order in the app.
4. Click on the "Get WooCommerce Shipping" button.
5. Click on "Add extension" to the store.
6. Confirm the pre-install screen is displayed with appropriate animation.

### Images/gif
https://user-images.githubusercontent.com/1657201/175008608-cb25d3fb-3480-49cc-9514-1a9c304876ed.mov
